### PR TITLE
Fetch Exercises Initial Draft

### PR DESCRIPTION
**Pull request by raulraja**
@dialelo Can you please review?

### CODE REVIEW

**Comment by dialelo** [comment]
Updated dependencies and went through the exercises making the tests pass. We can merge this and start preparing the inclusion of these exercises in Scala Exercises :tada: Take a look plz @raulraja

**Comment by raulraja** [comment]
This is great and I'm happy we finally found time to bring this up to the latest version. Since we both are part of this PR we should have @jdesiloniz or @juanpedromoreno give us their blessing.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+language: scala
+scala:
+  - 2.11.7
```
Comment (juanpedromoreno) [inline .travis.yml]: Should the scala version be 2.11.8 as the rest of libraries?

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,52 @@
+lazy val fetch = (project in file("."))
+.settings(publishSettings:_*)
+.enablePlugins(ExerciseCompilerPlugin)
+.settings(
+  organization := "org.scala-exercises",
+  name         := "exercises-fetch",
+  scalaVersion := "2.11.7",
```
Comment (juanpedromoreno) [inline build.sbt]: Same here, should the scala version be 2.11.8 as the rest of libraries?

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  libraryDependencies ++= Seq(
+    "com.fortysevendeg" %% "fetch" % "0.4.0",
+    "com.fortysevendeg" %% "fetch-monix" % "0.4.0",
+    "com.chuusai" %% "shapeless" % "2.2.5",
+    "org.scalatest" %% "scalatest" % "2.2.4",
+    "org.scala-exercises" %% "exercise-compiler" % version.value,
+    "org.scala-exercises" %% "definitions" % version.value,
+    "org.scalacheck" %% "scalacheck" % "1.12.5",
+    "com.github.alexarchambault" %% "scalacheck-shapeless_1.12" % "0.3.1",
+    compilerPlugin("org.spire-math" %% "kind-projector" % "0.7.1")
```
Comment (juanpedromoreno) [inline build.sbt]: Kind-projector -> 0.9.3? (fetch is using the latest version)

File: build.sbt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  )
+)
+
+// Distribution
+
+lazy val gpgFolder = sys.env.getOrElse("SE_GPG_FOLDER", ".")
```
Comment (juanpedromoreno) [inline build.sbt]: All the publish section should be aligned with the rest of the libraries, for example: https://github.com/scala-exercises/exercises-cats/blob/master/build.sbt#L27

SE_GPG_FOLDER  -> PGP_FOLDER
SE_GPG_PASSPHRASE -> PGP_PASSPHRASE
etc.

File: deploy.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
+
```
Comment (juanpedromoreno) [inline deploy.sh]: This file can be removed

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+jdk:
+  - oraclejdk8
+script:
+  - sbt test
+after_success:
+  - bash deploy.sh
```
Comment (juanpedromoreno) [inline .travis.yml]: Use this instead: https://github.com/scala-exercises/exercises-cats/blob/master/.travis.yml#L13

File: project/build.properties [hunk 7]
```diff
@@ -0,0 +1 @@
+sbt.version=0.13.9
```
Comment (juanpedromoreno) [inline project/build.properties]: -> 0.13.12 as the rest of the libraries

**Comment by dialelo** [comment]
Thanks for the review @juanpedromoreno, will update the PR and ping you as soon as it's ready.

**Comment by juanpedromoreno** [comment]
Great, thanks @dialelo!

**Comment by dialelo** [comment]
Take a last look and merge at your convenience @juanpedromoreno  :shipit:

**Pull request merged**